src/peer_discovery.cpp

```diff
@@ -80,11 +84,16 @@ PeerDiscovery::DomainPeerDiscovery::DomainPeerDiscovery(asio::ip::udp domain, in
     , sockFd_(*ioContext_, domain)
     , sockAddrSend_(asio::ip::address::from_string(domain.family() == AF_INET ? MULTICAST_ADDRESS_IPV4
                                                                               : MULTICAST_ADDRESS_IPV6), port)
-    , publishTimer_(*ioContext)
 {
-    sockFd_.set_option(asio::ip::multicast::join_group(sockAddrSend_.address()));
-    sockFd_.set_option(asio::ip::udp::socket::reuse_address(true));
-    sockFd_.bind({domain, port});
+    try {
+        sockFd_.set_option(asio::ip::multicast::join_group(sockAddrSend_.address()));
+        sockFd_.set_option(asio::ip::multicast::enable_loopback(false));
+        sockFd_.set_option(asio::ip::udp::socket::reuse_address(true));
+        sockFd_.bind({domain, port});
+    } catch (const std::exception& e) {
+        if (logger_)
+            logger_->e("%s failed %s", __PRETTY_FUNCTION__, e.what());
```

Comment:
Show proper error:
`logger_->e("Can't start peer discovery for %s: %s", domain.family() == AF_INET ? "IPv4" : "IPv6", e.what());`